cache/cache.go

```diff
@@ -1,65 +1,58 @@
 package cache
 
 import (
+	"io"
 	"os"
 	"path/filepath"
+	"strings"
 
-	bolt "github.com/simar7/gokv/bbolt"
-	"github.com/simar7/gokv/encoding"
-	kvtypes "github.com/simar7/gokv/types"
 	"golang.org/x/xerrors"
 )
 
+var (
+	replacer = strings.NewReplacer("/", "_")
+)
+
 type Cache interface {
-	Get(bucket, key string, value *[]byte) (found bool, err error)
-	Set(bucket, key string, value []byte) (err error)
+	Get(key string) io.Reader
+	Set(key string, file io.Reader) (io.Reader, error)
 	Clear() error
 }
 
-type RealCache struct {
-	directory string
-	cache     *bolt.Store
+type FSCache struct {
+	Directory string
 }
 
-func New(cacheDir string) (Cache, error) {
-	dir := filepath.Join(cacheDir, "fanal")
-	if err := os.MkdirAll(dir, 0700); err != nil {
-		return nil, xerrors.Errorf("unable to create cache dir: %w", err)
-	}
-
-	cacheOptions := bolt.Options{
-		RootBucketName: "fanal",
-		Path:           filepath.Join(dir, "cache.db"),
-		Codec:          encoding.Raw,
-	}
+func Initialize(cacheDir string) Cache {
+	return &FSCache{Directory: cacheDir}
```

Comment:
```suggestion
	return &FSCache{Directory: filepath.Join(cacheDir, "fanal")}
```

`cacheDir` is usually a system cache directory itself. So, in my environment, it is `/Users/teppei/Library/Caches/`. If we scan `alpine:3.10`, it will be stored in `/Users/teppei/Library/Caches/alpine:3.10`. We should add one more directory to store image caches. The system cache directory will be deleted with `--clear` option if we don't add one more directory.

Besides, we should define it in const like the below.

```
const(
    cacheDirName = "fanal"
)
```